File: src/app/interfaces/receipt-approval-dto.ts
import { ReceiptItemsApprovalDTO } from "./receipt-items-approval-dto";

export interface ReceiptApprovalDTO {
    receiptId: string;
    approvedBy: string;
    userId: string;
    approvedOn: string;
    merchantId: number;
    merchantName: string;
    merchantAddress: string;
    merchantPhone: string;
    merchantEmail: string;
    customerName: string;
    customerAddress: string;
    customerPhone: string;
    receiptNumber: string;
    receiptDate: string;
    subTotal: number;
    taxAmount: number;
    serviceCharge: number;
    otherCharge: number;
    statusId: number;
    totalAmount: number;
    discount: number;
    receiptItemsApproval: ReceiptItemsApprovalDTO[];
}

File: API/FunctionAppDoc2Data/Models/ReceiptApprovalDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FunctionAppDoc2Data.Models;
public class ReceiptApprovalDTO
{
    public Guid ReceiptId { get; set; }
    public Guid ApprovedBy { get; set; }
    public Guid UserId { get; set; }
    public DateTime ApprovedOn { get;}  = DateTime.UtcNow;
    public int MerchantId { get; set; }
    public string MerchantName { get; set; }
    public string MerchantAddress { get; set; }
    public string MerchantPhone { get; set; }
    public string MerchantEmail { get; set; }
    public string CustomerName { get; set; }
    public string CustomerAddress { get; set; }
    public string CustomerPhone { get; set; }
    public string ReceiptNumber { get; set; }
    public DateTime ReceiptDate { get; set; }
    public decimal SubTotal { get; set; }
    public decimal TaxAmount { get; set; }
    public decimal ServiceCharge { get; set; } = 0;
    public decimal OtherCharge { get; set; } = 0;
    public int StatusId { get; set; }
    public decimal TotalAmount { get; set; }
    public decimal Discount { get; set; } = 0;
    public List<ReceiptItemsApprovalDTO> ReceiptItemsApproval { get; set; }

}
